src/main/java/org/folio/domain/ActionBlocks.java

```diff
@@ -22,68 +23,83 @@
 public class ActionBlocks {
   private static final Logger log = LogManager.getLogger(ActionBlocks.class);
   private static final Double NUMBER_OF_MINUTES_IN_ONE_DAY = 1440.0;
+  private static final String LOG_TEMPLATE_BY_LIMIT_CONDITION = "byLimit:: condition is {}";
 
   private final boolean blockBorrowing;
   private final boolean blockRenewals;
   private final boolean blockRequests;
 
-  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit limit) {
-    if (userSummary == null || limit == null || userSummary.getOpenLoans() == null) {
-      log.error("Failed to determine blocks: one of the parameters is null");
+  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit patronBlockLimit) {
+    log.debug("byLimit:: parameters userSummary: {}, patronBlockLimit: {}",
+      () -> logAsJson(userSummary), () -> patronBlockLimit);
+    if (userSummary == null || patronBlockLimit == null || userSummary.getOpenLoans() == null) {
+      log.warn("byLimit:: Failed to determine blocks because one of the parameters is null; " +
+        "parameters userSummary: {}, patronBlockLimit: {}", () -> logAsJson(userSummary),
+        () -> patronBlockLimit);
       return empty();
     }
-
-    return byLimit(userSummary, limit, userSummary.getOpenLoans().stream()
+    ActionBlocks actionBlocks = byLimit(userSummary, patronBlockLimit, userSummary.getOpenLoans().stream()
       .filter(openLoan -> openLoan.getLoanId() != null)
       .collect(Collectors.toMap(OpenLoan::getLoanId, r -> 0)));
+    log.info("byLimit:: result: {}", () -> logAsJson(actionBlocks));
+    return actionBlocks;
   }
 
-  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit limit,
+  public static ActionBlocks byLimit(UserSummary userSummary, PatronBlockLimit patronBlockLimit,
     Map<String, Integer> overdueMinutes) {
-
-    if (userSummary == null || limit == null || overdueMinutes == null || limit.getValue() == null
-      || limit.getConditionId() == null) {
-
-      log.error("Failed to determine blocks: one of the parameters is null");
+    log.debug("byLimit:: parameters userSummary: {}, patronBlockLimit: {}, " +
+      "overdueMinutes: {}", () -> logAsJson(userSummary), () -> patronBlockLimit,
+      () -> overdueMinutes);
+    if (userSummary == null || patronBlockLimit == null || overdueMinutes == null ||
+      patronBlockLimit.getValue() == null || patronBlockLimit.getConditionId() == null) {
+      log.warn("byLimit:: Failed to determine blocks because one of the parameters is null; " +
+        "parameters userSummary: {}, patronBlockLimit: {}, patronBlockLimit.value, " +
```

Comment:
patronBlockLimit.value, patronBlockLimit.conditionId

Values for these properties are not logged?

Reply:
We're logging the whole object patronBlockLimit